docs/bugs.json

```diff
@@ -1,4 +1,14 @@
 [
+    {
+        "uid": "SOL-2022-2",
+        "name": "NestedCallataArrayAbiReencodingSizeValidation",
+        "summary": "ABI-encoding values in calldata that contain nested dynamic arrays sometimes read beyond ``calldatasize()`` instead of reverting.",
+        "description": "Calldata validation for nested dynamic types is deferred until the first access to the nested values. Such an access may for example be a copy to memory or an index or member access to the outer type. While in most such accesses calldata validation correctly checks that the data area of the nested array is completely contained in the passed calldata (i.e. in the range [0, calldatasize()]), this check may fail, when ABI encoding such nested types again directly from calldata. For instance, this can, if a value in calldata with a nested dynamic array is passed to an external call, used in ``abi.encode`` or emitted as event. In such cases, if the data area of the nested array extends beyond ``calldatasize()``, ABI encoding it did not revert, but continue reading values from beyond ``calldatasize()`` (i.e. zero values).",
+        "link": "TODO",
+        "introduced": "0.5.8",
+        "fixed": "0.8.14",
+        "severity": "very low"
```

Comment:
Any opinions on the severity? Is ``very low`` an underestimation?
In most cases, I'd expect little harm to come from reading zeroes from calldata, but it may be worth thinking about whether weird cases can happen...
Also worth a thought whether the inconsistency in the revert behaviour depending on directly accessing or re-encoding calldata has any serious implications for proxy patterns or the like.

Reply:
On the other hand one can also argue for the direct effect of the bug, that this (in certain cases) can undermine the safety guard against transactions with corrupted calldata... based on that one could also go as high as ``medium``...

Reply:
@chriseth in channel: I think it's very low